src/main/java/org/javarosa/xform/parse/XFormParser.java

```diff
@@ -471,9 +474,8 @@ private void parseDoc(Map<String, String> namespacePrefixesByUri, String lastSav
                 final String instanceId = instanceNodeIdStrs.get(instanceIndex);
                 final String instanceSrc = parseInstanceSrc(instance, lastSavedSrc);
 
-                // Disable jr://file-csv/ support by explicitly only supporting jr://file/
-                // until https://github.com/opendatakit/javarosa/issues/417 is addressed
-                if (instanceSrc != null && instanceSrc.toLowerCase().startsWith("jr://file/")) {
+                // only build when ESI Id is not referenced in pulldata
+                if (instanceSrc != null && instanceSrc.toLowerCase().startsWith("jr://file") && secondaryInstanceAnalyzer.shouldSecondaryInstanceBeParsed(instanceId)) {
```

Comment:
This will always be false because the primary instance is loaded before the secondary instances are, right? Doesn't `analyzeElement` always happen after this has already been done?